File: kahve_qr_olusturucu_flutter-calismalarim/lib/usecases/IndirimYuzdeHesapla.dart
String indirimYuzdeHesapla(double indirimsizFiyat, double indirimliFiyat) {
  if (indirimsizFiyat <= 0) return "0.0";

  double indirimOrani = ((indirimsizFiyat - indirimliFiyat) / indirimsizFiyat) * 100;
  return indirimOrani.toStringAsFixed(1);
}

File: KahveQRKodOlusturucuJetpack-calismalarim/app/src/main/java/com/example/qrkodolusturucujetpack/usecases/IndirimYuzdeHesapla.kt
package com.example.qrkodolusturucujetpack.usecases

fun indirimYuzdeHesapla(indirimsizFiyat: Double, indirimliFiyat: Double): String {
    if (indirimsizFiyat <= 0) return "0.0"

    val indirimOrani = ((indirimsizFiyat - indirimliFiyat) / indirimsizFiyat) * 100
    return String.format("%.1f", indirimOrani)
}
